### ISSUE TITLE
Support for Arduino Core 3.1.0 / IDF 5.3.2

### ISSUE THREAD

**Issue by proddy** [b5i/backtrace-saver-esp32#1]
nice library. I couldn't however get it to compile with the latest Arduino Core 3.1.0  and IDF 5.3.2. It complained about missing soc header files. Just wanted to point this out. If I have time I'll dig deeper and try and resolve it.

**Comment by b5i** [comment]
Hi! Thank you for pointing out this issue, if you aren't able to solve it do not hesitate to give me the compilation command, its output and eventually a way to reproduce the bug (a repo to test for example).

**Comment by proddy** [comment]
Yeah, sorry, my post wasn't very useful. So, as you probably know, PlatformIO had an argument with Espressif and dropped all support beyond Arduino Core 2.0.18 and IDF 4.4.x. The latest version of Arduino Core is 3.1 and uses IDF 5.3. If I change the `platform` option in `platformio.ini` to `platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip`, it will download the latest libraries and matching toolchain, but you'll see the compile errors. Maybe I need to add `espidf` to the PIO `framework` option. I didn't explore further at this point.

**Comment by b5i** [comment]
It seems like a new `set_arduino_panic_handler` has to be used, I'll try to implement it.

**Comment by b5i** [comment]
The compatibility issue should be resolved in `main`, could you check if it's working for you please?

**Comment by proddy** [comment]
will do, thanks!

**Comment by i-am-shodan** [comment]
I'm using 3.3.0 and I can't seem to get this to work. I'm calling `backtrace_saver::init();` and `backtrace_saver::debugBacktraceUpdate()` but the stack trace is always empty after a reset

**Comment by i-am-shodan** [comment]
I've also tried adding a #define to BT_SAVER_NEW_API

**Comment by b5i** [comment]
Hello, can you provide a minimal working example demonstrating the bug please? Thank you
